### PR TITLE
avatars, resolves #1875

### LINKED ISSUE #1875: avatars on project tiles
- looking at self
- looking at someone else’s avatar
- “v”’s (360) avatar
- looking at missing avatar
- looking at self among other tiles (e.g. explore)
- missing avatar among other tiles

also related #1716

### PR DESCRIPTION
**Pull request by gmarinov**
- adds default avatar for all registered users
 - adds 360.vizor.io avatar
 - adds /examples avatar
 - deprecates gravatars

screen shot 2016-09-08 at 23 38 27

### CODE REVIEW

File: services/assetService.js [hunk 1]
```diff
@@ -200,5 +201,27 @@ AssetService.prototype.save = function(data, user) {
 	})
 }
 
+AssetService.prototype.getOwnersInfo = function(/* Array */ creatorIds) {
```
Comment (kschzt) [inline services/assetService.js]: this should probably live in the User model? `findUsersById(ids)`

File: services/graphService.js [hunk 2]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
...
+ * @return {list, meta, owners}
+ * @private
+ */
+GraphService.prototype._listWithOwners = function(filter, sort, select, paging) {
+	var that = this
+	var listData=[]
+	return this.countAndFind(filter, sort, select, paging)
+		.then((data) => {
+			listData = data
+			var creatorIds = {}
```
Comment (kschzt) [inline services/graphService.js]: turn this into an array, `creatorIds = []`

File: services/graphService.js [hunk 3]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
...
+			// only interested in unique ids
+			for (var row of data.list) {
+				if (row._creator)
+					creatorIds[row._creator] = true
```
Comment (kschzt) [inline services/graphService.js]: `creatorIds.push(row._creator)`

File: services/graphService.js [hunk 4]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
...
+			}
+			return that.getOwnersInfo(Object.keys(creatorIds))
```
Comment (kschzt) [inline services/graphService.js]: `User.findUsersByIds(creatorIds)`

File: services/graphService.js [hunk 5]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
 	})
 }
 
+/**
+ * @param paging
+ * @return {list, meta, owners}
+ * @private
+ */
+GraphService.prototype._listWithOwners = function(filter, sort, select, paging) {
```
Comment (kschzt) [inline services/graphService.js]: please add an integration test for this.

File: services/graphService.js [hunk 6]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
...
+		})
+		.then((ownersInfo) => {
+			listData.owners = ownersInfo
+			return Promise.resolve(listData)
+		})
+}
+
+
+// admin
 GraphService.prototype.listWithPreviews = function(paging) {
```
Comment (kschzt) [inline services/graphService.js]: please add an integration test for this, rather

File: services/graphService.js [hunk 7]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
...
+			return that.getOwnersInfo(Object.keys(creatorIds))
+		})
+		.then((ownersInfo) => {
+			listData.owners = ownersInfo
+			return Promise.resolve(listData)
+		})
+}
+
+
+// admin
```
Comment (kschzt) [inline services/graphService.js]: admin what?

Comment (kschzt) [inline services/graphService.js]: oh, `/admin/list`

File: services/graphService.js [hunk 8]
```diff
@@ -28,19 +28,45 @@ GraphService.prototype.findByPath = function(path) {
...
 }
 
 GraphService.prototype.publicRankedList = function(paging) {
 	var filter = { private: false, deleted: false }
 	var sort = {'rank': -1, 'updatedAt': -1}
-	var select = '_creator private owner name previewUrlSmall updatedAt stat rank'
-
-	return this.countAndFind(filter, sort, select, paging)
+	var select = '_creator private owner name previewUrlSmall updatedAt stat rank views'
+	return this._listWithOwners(filter, sort, select, paging)
```
Comment (kschzt) [inline services/graphService.js]: is there already a test for this? I'd like to have a test added any time the backend is changed.

File: services/graphService.js (hunk 4 above)
Comment (gmarinov) [inline services/graphService.js]: @kschzt but creatorIds are not unique if an array, which is why i had them into a {} so that keys overwrite each other

**Pull request merged**